app/components/settings/project_custom_field_sections/show_component.rb

```diff
@@ -0,0 +1,127 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Settings
+  module ProjectCustomFieldSections
+    class ShowComponent < ApplicationComponent
+      include ApplicationHelper
+      include OpPrimer::ComponentHelpers
+      include OpTurbo::Streamable
+
+      def initialize(project_custom_field_section:)
+        super
+
+        @project_custom_field_section = project_custom_field_section
+        @project_custom_fields = project_custom_field_section.custom_fields.reorder(position_in_custom_field_section: :asc)
+      end
+
+      private
+
+      def wrapper_uniq_by
+        @project_custom_field_section.id
+      end
+
+      def drag_and_drop_target_config
+        {
+          'is-drag-and-drop-target': true,
+          'target-container-accessor': '.Box > ul', # the accessor of the container that contains the drag and drop items
+          'target-id': @project_custom_field_section.id, # the id of the target
+          'target-allowed-drag-type': 'custom-field' # the type of dragged items which are allowed to be dropped in this target
+        }
+      end
+
+      def draggable_item_config(project_custom_field)
+        {
+          'draggable-id': project_custom_field.id,
+          'draggable-type': 'custom-field',
+          'drop-url': drop_admin_settings_project_custom_field_path(project_custom_field)
+        }
+      end
+
+      def move_actions(menu)
+        # TODO: these methods trigger database queries for each section displayed
```

Comment:
Yes, this will trigger N+1 queries. I have solved a similar issue on the meeting agenda items in this PR #14190 . 
The solution is to pass down the first and last element information to the ShowComponent in order to avoid the N+1 queries.

Reply:
Uh nice! Would like to implement here as well? :)